gems/sorbet-runtime/lib/types/private/methods/_methods.rb

```diff
@@ -180,6 +182,16 @@ def self.note_module_deals_with_final(mod)
     @modules_with_final[mod.singleton_class]
   end
 
+  def self._with_declaring_final_method(&blk)
```

Comment:
Do we want the interface to be a block? or to be forced to repeat something for the next method?

like `_make_next_method_final`